ScheduleGenerator/validator.scala

```diff
@@ -0,0 +1,265 @@
+        throw new ValidationException(
+          "The number of  partitions does not match the number of partitions");
+
+      // Create a list of time entities
+      var entities: List[TimeEntity] = this.createTimeEntities();
+
+      // Create "schedulability" table and check for overlapping schedules - return the configuration
+      return this.createTable(entities);
+    }
+
+    // Done using inspiration from this lecture: https://www.moodle.aau.dk/pluginfile.php/1163980/mod_resource/content/10/Lecture-4.pdf (Overlapping activity schedule problem)
+    def createTable(entities: List[TimeEntity]): String = {
+      // Initialise matrix of the matrix configuration and core list table (allocate space for all possible schedules)
+      // This initialises a table of Affinity[Entities[Values]]
+      var scheduleTable: Array[Array[Array[Integer]]] = Array(
+        Array.ofDim[Integer](entities.length, this.TABLE_HEIGHT),
+        Array.ofDim[Integer](entities.length, this.TABLE_HEIGHT),
+        Array.ofDim[Integer](entities.length, this.TABLE_HEIGHT),
+        Array.ofDim[Integer](entities.length, this.TABLE_HEIGHT)
+      );
+
+      // Sort entities list based on when periods start (Default ascending based on offsets)
+      var sortedEntities = entities.sortBy(ent => ent.offset);
+
+      // Iterate through the entity table
+      for (ent <- sortedEntities) {
+        // Check for overlapping periods
+        scheduleTable = this.generateAndCheckSchedule(ent, scheduleTable);
+      }
+
+      return printScheduleConfigation(entities, scheduleTable);
+    }
+
+    def printScheduleConfigation(
+        entities: List[TimeEntity],
+        scheduleTable: Array[Array[Array[Integer]]]): String = {
+      var emptyField = false;
+      var cString = "|";
+      for (c <- 0 to this.TOTAL_CORES - 1) {
+        cString = cString + f"---------------------------------------------------------------------|\n";
+        cString = cString + f"|·············|·············|·····${c}·······|·············|·············|\n";
+        cString = cString + f"| identifier  |   period    |  duration   |    offset   |  affinity   |\n";
+        cString = cString + f"|---------------------------------------------------------------------|\n|";
+        for (i <- 0 to entities.length - 1) {
+          for (j <- 0 to this.TABLE_HEIGHT - 1) {
+            if (scheduleTable(c)(i)(j) != null) {
+              cString = cString + f"${this.mkRow(scheduleTable(c)(i)(j), 13)}|";
+            } else {
+              emptyField = true;
+            }
+          }
+
+          if (emptyField) {
+            cString = cString + f"             |             |             |             |             |\n|";
+            emptyField = false;
+          } else {
+            cString = cString + "\n|";
+          }
+        }
+      }
+      cString = cString + f"---------------------------------------------------------------------|\n";
+
+      return cString;
```

Comment:
TODO: Maybe use stringbuilder instead of string append (+) to reduce execution time. 
This might not be a problem.